content/documentation/pages/9-applications/0-index.md

```diff
@@ -0,0 +1,85 @@
+
+<!--TIP-->
+
+The URL may be any standard URL or may use one of the Data Flow `maven://` or `docker://` formats described in [Stream Processing with Data Flow](%currentPath%/stream-developer-guides/streams/data-flow-stream/#application-registration).
+
+<!--END_TIP-->
+
+To optimize performance, you may package [application metadata](%currentPath%/feature-guides/general/application-metadata), which contains the names and descriptions of exposed application properties, in a separate smaller artifact. This is not required but, since the metadata is typically accessed before the application binary is needed, it allows more efficient use of network resources when using Data Flow. In this case, add a registration entry for the metadata artifact as `<type>.<name>.metadata=app-metadata-url`.
+
+Here is a snippet of a bulk registration file used to register Maven artifacts:
+
+```
+sink.cassandra=maven://org.springframework.cloud.stream.app:cassandra-sink-rabbit:2.1.2.RELEASE
+sink.cassandra.metadata=maven://org.springframework.cloud.stream.app:cassandra-sink-rabbit:jar:metadata:2.1.2.RELEASE
+sink.s3=maven://org.springframework.cloud.stream.app:s3-sink-rabbit:2.1.2.RELEASE
+sink.s3.metadata=maven://org.springframework.cloud.stream.app:s3-sink-rabbit:jar:metadata:2.1.2.RELEASE
+```
+
+From the Data Flow Shell, you can bulk register the applications:
+
+```bash
+dataflow:>app import --uri file://path-to-my-app-registration.properties
+```
+
+You can also pass the --local option (which is true by default) to indicate whether the properties file location should be resolved within the shell process itself. If the location should be resolved from the Data Flow Server process, specify --local false.
+
+<!--CAUTION-->
+
+When using either `app register` or `app import`, if an app is already registered with the provided name and type and version, it is not overridden by default. If you would like to override the pre-existing app uri or metadata-uri coordinates, then include the --force option.
+
+Note, however, that, once downloaded, applications may be cached locally on the Data Flow server, based on the resource location. If the resource location does not change (even though the actual resource bytes may be different), then it is not re-downloaded. When using maven:// resources on the other hand, using a constant location may still circumvent caching (if using -SNAPSHOT versions).
+
+Moreover, if a stream is already deployed and using some version of a registered app, then (forcibly) re-registering a different app has no effect until the stream is deployed again.
+
+<!--END_CAUTION-->
+
+<!--TIP-->
+
+In some cases, the Resource is resolved on the server side. In others, the URI is passed to a runtime container instance where it is resolved. Consult the specific documentation of each Data Flow Server for more detail.
```

Comment:
Can we be more specific about this? Also, `documentation of each Data Flow Server` is confusing.

Reply:
I'm removing this. I'm not sure what it means and is not really helpful.